### PR TITLE
Feature/services connection

### PR DESCRIPTION
**Pull request by AlbertPG**

### CODE REVIEW

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 13
           

Coverage decreased per file
===========================
- app/src/main/java/org/pesmypetcare/mypetcare/features/users/UserNotExistingException.java  -100
         

Complexity increasing per file
==============================
- app/src/main/java/org/pesmypetcare/mypetcare/services/PetManagerAdapter.java  3
- app/src/main/java/org/pesmypetcare/mypetcare/services/ServiceLocator.java  2
- app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/ImageZoomFragment.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/NewPasswordFragment.java  3
- app/src/main/java/org/pesmypetcare/mypetcare/activities/views/PetEntryView.java  1
- app/src/androidTest/java/org/pesmypetcare/mypetcare/activities/fragments/TestInfoPetFragment.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/services/UserManagerAdapter.java  1
- app/src/androidTest/java/org/pesmypetcare/mypetcare/activities/fragments/TestImageZoomFragment.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/utilities/DateConversion.java  2
         

Complexity decreasing per file
==============================
+ app/src/main/java/org/pesmypetcare/mypetcare/controllers/TrObtainUser.java  -2
         

Clones added
============
- app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/InfoPetFragment.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/RegisterPetFragment.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5254198&bid=17216395)

**Review by marclll — CHANGES_REQUESTED** [review]
It doesn't pass the tests

**Review by xavier-campos-diaz — APPROVED** [review]

**Review by enrichp — APPROVED** [review]

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 12
- Added 1
           

Coverage decreased per file
===========================
- app/src/main/java/org/pesmypetcare/mypetcare/features/users/UserNotExistingException.java  -100
         

Complexity increasing per file
==============================
- app/src/androidTest/java/org/pesmypetcare/mypetcare/activities/fragments/TestInfoPetFragment.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/NewPasswordFragment.java  3
- app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/ImageZoomFragment.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/utilities/DateConversion.java  2
- app/src/main/java/org/pesmypetcare/mypetcare/services/PetManagerAdapter.java  3
- app/src/androidTest/java/org/pesmypetcare/mypetcare/activities/fragments/TestImageZoomFragment.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/activities/views/PetEntryView.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/services/UserManagerAdapter.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/services/ServiceLocator.java  2
         

Complexity decreasing per file
==============================
+ app/src/main/java/org/pesmypetcare/mypetcare/controllers/TrObtainUser.java  -2
         

Clones added
============
- app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/InfoPetFragment.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/RegisterPetFragment.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5254198&bid=17216395)

File: app/src/androidTest/java/org/pesmypetcare/mypetcare/activities/fragments/TestInfoPetFragment.java [hunk 1]
```diff
@@ -0,0 +1,97 @@
+package org.pesmypetcare.mypetcare.activities.fragments;
+
+import android.view.View;
+
```
Comment (santidrj) [inline app/src/androidTest/java/org/pesmypetcare/mypetcare/activities/fragments/TestInfoPetFragment.java]: Codacy Issue found: [Unused import - org.junit.Before.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5254198)

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 12
           

Coverage decreased per file
===========================
- app/src/main/java/org/pesmypetcare/mypetcare/features/users/UserNotExistingException.java  -100
         

Complexity increasing per file
==============================
- app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/ImageZoomFragment.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/NewPasswordFragment.java  3
- app/src/main/java/org/pesmypetcare/mypetcare/activities/views/PetEntryView.java  1
- app/src/androidTest/java/org/pesmypetcare/mypetcare/activities/fragments/TestImageZoomFragment.java  1
- app/src/androidTest/java/org/pesmypetcare/mypetcare/activities/fragments/TestInfoPetFragment.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/services/ServiceLocator.java  2
- app/src/main/java/org/pesmypetcare/mypetcare/services/PetManagerAdapter.java  3
- app/src/main/java/org/pesmypetcare/mypetcare/utilities/DateConversion.java  2
- app/src/main/java/org/pesmypetcare/mypetcare/services/UserManagerAdapter.java  1
         

Complexity decreasing per file
==============================
+ app/src/main/java/org/pesmypetcare/mypetcare/controllers/TrObtainUser.java  -2
         

Clones added
============
- app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/RegisterPetFragment.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/InfoPetFragment.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5254198&bid=17216395)

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 12
           

Coverage decreased per file
===========================
- app/src/main/java/org/pesmypetcare/mypetcare/features/users/UserNotExistingException.java  -100
         

Complexity increasing per file
==============================
- app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/NewPasswordFragment.java  3
- app/src/main/java/org/pesmypetcare/mypetcare/services/UserManagerAdapter.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/services/PetManagerAdapter.java  3
- app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/ImageZoomFragment.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/activities/views/PetEntryView.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/services/ServiceLocator.java  2
- app/src/main/java/org/pesmypetcare/mypetcare/utilities/DateConversion.java  2
- app/src/androidTest/java/org/pesmypetcare/mypetcare/activities/fragments/TestImageZoomFragment.java  1
- app/src/androidTest/java/org/pesmypetcare/mypetcare/activities/fragments/TestInfoPetFragment.java  1
         

Complexity decreasing per file
==============================
+ app/src/main/java/org/pesmypetcare/mypetcare/controllers/TrObtainUser.java  -2
         

Clones added
============
- app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/InfoPetFragment.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/RegisterPetFragment.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5254198&bid=17216395)

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 12
           

Coverage decreased per file
===========================
- app/src/main/java/org/pesmypetcare/mypetcare/features/users/UserNotExistingException.java  -100
         

Complexity increasing per file
==============================
- app/src/main/java/org/pesmypetcare/mypetcare/services/PetManagerAdapter.java  3
- app/src/main/java/org/pesmypetcare/mypetcare/activities/views/PetEntryView.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/NewPasswordFragment.java  3
- app/src/main/java/org/pesmypetcare/mypetcare/utilities/DateConversion.java  2
- app/src/main/java/org/pesmypetcare/mypetcare/services/ServiceLocator.java  2
- app/src/androidTest/java/org/pesmypetcare/mypetcare/activities/fragments/TestInfoPetFragment.java  1
- app/src/androidTest/java/org/pesmypetcare/mypetcare/activities/fragments/TestImageZoomFragment.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/ImageZoomFragment.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/services/UserManagerAdapter.java  1
         

Complexity decreasing per file
==============================
+ app/src/main/java/org/pesmypetcare/mypetcare/controllers/TrObtainUser.java  -2
         

Clones added
============
- app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/InfoPetFragment.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/RegisterPetFragment.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5254198&bid=17216395)

**Pull request merged**